src/realm/parser/parser_utils.hpp

```diff
@@ -105,7 +105,8 @@ bool try_parse_specials(std::string str, T& ret)
         ret = std::numeric_limits<T>::infinity();
         return true;
     }
-    else if (std::numeric_limits<T>::has_infinity && (str == "-infinity" || str == "-inf")) {
+    else if (std::numeric_limits<T>::has_infinity && std::numeric_limits<T>::is_signed &&
```

Comment:
I don't think there are any unsigned types that have infinity. Really, I think this could all just be `std::numeric_limits<T>::is_iec559` since we only care about IEEE754 (the more common name for iec559) float point types, and they all have sign bits, nans and infs.

Reply:
The documentation I found only says that `::is_iec559` "may be true" for float and double. I didn't feel like tracking down the cases where they are not true, so I created a more inclusive function which will enable the special checks explicitly for float, double, and any other types for which `::is_iec559` is true.